src/hapi/pipelines/database/operational_presence.py

```diff
@@ -82,64 +87,65 @@ def populate(self, debug=False):
                             add_message(
                                 errors,
                                 dataset_name,
-                                f"org {org_name_orig} missing sector",
+                                f"org {org_str} missing sector",
                             )
                             continue
-                        org_type_orig = None
-                        if org_type_name_index:
-                            org_type_orig = values[org_type_name_index][
-                                admin_code
-                            ][i]
-                        country_code = admin_code
+                        sector_code = self._sector.get_sector_code(sector_orig)
+                        if not sector_code:
+                            add_missing_value_message(
+                                errors, dataset_name, "sector", sector_orig
+                            )
+                            continue
+
+                        # * Admin processing
                         if admin_level == "admintwo":
                             country_code = self._admintwo.pcode_to_iso3.get(
                                 admin_code
                             )
-                        if admin_level == "adminone":
+                        elif admin_level == "adminone":
                             country_code = self._adminone.pcode_to_iso3.get(
                                 admin_code
                             )
-                        org_info = self._org.get_org_info(
-                            org_name_orig, location=country_code
+                        else:
+                            country_code = admin_code
+                        admin2_code = admins.get_admin2_code_based_on_level(
+                            admin_code=admin_code, admin_level=admin_level
                         )
-                        org_name = org_info.get("#org+name")
-                        self._org.add_org_to_lookup(org_name_orig, org_name)
-                        org_acronym = org_info.get(
-                            "#org+acronym",
-                            values[org_acronym_index][admin_code][i],
+                        admin2_ref = self._admins.admin2_data[admin2_code]
+
+                        # * Org processing
+                        if not org_str:
+                            org_str = values[org_acronym_index][admin_code][i]
+                        org_info = self._org.get_org_info(
+                            org_str, location=country_code
                         )
-                        if org_acronym is not None and len(org_acronym) > 32:
-                            org_acronym = org_acronym[:32]
-                        org_type_code = org_info.get("#org+type+code")
-                        org_type_name = None
-                        if not org_type_code:
-                            if org_type_name_index:
+                        if not org_info.complete:
```

Comment:
Orgs require acronyms but not org types right? Should they be treated differently? e.g. here if there is an org without a type, you will still be checking it on every iteration. But perhaps that is the right thing to do.

Reply:
I think it's the right thing to do - then if one row in the data has a type we can fill in that type in the org table

Reply:
Yes i think it must be checked until the type is found or until there is nothing left to check